### PR TITLE
Refactor data_utils

### LINKED ISSUE #186: Code cleanup for utils/data_utils
Move all of the `data_loaders` into load_utils
move `combine_xarrays` and combine_point_dirs into misc_utils

Delete crop_image_stack and crop_helper (this functionality has moved to deepcell)

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

A mini-PR to refactor data_utils. Addresses and closes #186.

**How did you implement your changes**

We add the loader functions into a separate file, `load_utils.py`, and functions that don't really fall under the realm of `data_utils.py` and `load_utils.py` into `misc_utils.py`.

**Remaining issues**

It doesn't appear that we're using either `combine_xarrays` or `combine_point_directories` anywhere else but `misc_utils.py` and `misc_utils_test.py` themselves. Should we still keep them around?

### CODE REVIEW

File: ark/utils/data_utils.py [hunk 1]
```diff
@@ -367,194 +39,6 @@ def generate_deepcell_input(data_xr, data_dir, nuc_channels, mem_channels):
...
-                cropped_images = crop_helper(image_stack, crop_size)
-            else:
-                # crop the image by the shift prior to generating grid of crops
-                img_shift = image_stack[:, (row_shift * stride_step):,
-                                        (col_shift * stride_step):, :]
-                # print("shape of the input image is {}".format(img_shift.shape))
-                temp_images = crop_helper(img_shift, crop_size)
-                cropped_images = np.append(cropped_images, temp_images, axis=0)
-
-    return cropped_images
```
Comment (ackagel) [inline ark/utils/data_utils.py:529]: gone but not forgotten

File: ark/utils/data_utils_test.py [hunk 2]
```diff
@@ -261,91 +71,6 @@ def test_generate_deepcell_input():
...
-    # test with overlap
-    crop_input = np.zeros((4, 256, 256, 4), dtype="int16")
-    crop_size = 64
-    stride_fraction = 0.25
-
-    cropped = data_utils.crop_image_stack(crop_input, crop_size, stride_fraction)
-    num_crops = crop_input.shape[0] * (crop_input.shape[1] // crop_size) * \
-        (crop_input.shape[2] // crop_size) * (1 / stride_fraction) * (1 / stride_fraction)
-
-    assert np.array_equal(cropped.shape, (num_crops, crop_size, crop_size, crop_input.shape[3]))
```
Comment (ackagel) [inline ark/utils/data_utils_test.py:324]: used to test with the best 😭

**Review by ackagel — COMMENTED** [review]
definitely good to split up data utils; it was getting mega big.  everything looks okay, but there's about >100 lines that didn't survive the transition from `data_utils` to either `load_utils` or `misc_utils`.  It looks like that was `crop_helper` and `crop_image_stack` and their test functions.  any reason these were deleted?

**Comment by alex-l-kong** [comment]
> definitely good to split up data utils; it was getting mega big. everything looks okay, but there's about >100 lines that didn't survive the transition from `data_utils` to either `load_utils` or `misc_utils`. It looks like that was `crop_helper` and `crop_image_stack` and their test functions. any reason these were deleted?

The issue stated that the functionality of `crop_image_stack` and `crop_helper` was moved to deepcell's end.

File: ark/utils/data_utils.py (hunk 1 above)
Comment (alex-l-kong) [reply ark/utils/data_utils.py:529]: See comment below

File: ark/utils/data_utils_test.py (hunk 2 above)
Comment (alex-l-kong) [reply ark/utils/data_utils_test.py:324]: Ditto

**Comment by ackagel** [comment]
my b, looks good

**Review by ackagel — APPROVED** [review]

File: ark/utils/data_utils_test.py (hunk 2 above)
Comment (ngreenwald) [reply ark/utils/data_utils_test.py:324]: Yeah RIP, I saw you added a bunch of testing for these functions in one of your PRs

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**